pure.zsh

```diff
@@ -292,6 +292,16 @@ prompt_pure_async_git_fetch() {
 	# Set SSH `BachMode` to disable all interactive SSH password prompting.
 	export GIT_SSH_COMMAND="${GIT_SSH_COMMAND:-"ssh"} -o BatchMode=yes"
 
+	local ref
+	ref=$(command git symbolic-ref -q HEAD)
+	local -a remote
+	remote=($(command git for-each-ref --format='%(upstream:remotename) %(refname)' $ref))
```

Comment:
I'm not an expert in git or Pure but perhaps this would do what you need?

```zsh
local remote="$(command git rev-parse --abbrev-ref --symbolic-full-name @{u} 2>/dev/null)"
```

Reply:
I did consider this, on the one hand, we wouldn't need two git commands (`symbolic-ref` + `for-each-ref`), on the other we avoid parsing the output. With the above command we'd need to split that string on `/` for it to work with `git fetch` (the difference here would be `origin/master` vs `origin master`). And we'd have to consider other aspects like can origin contain `/`? and are we properly handling branches that contain `/`.

But I'd like to understand if these two commands actually produce different output or not (I'm no git expert myself). If the output is the same (ignoring `/` vs space), then it won't at least help with Sindres issue.

Reply:
Good point about parsing ambiguity. I think both branches and remotes can have `/` in them, so there is no way to unambiguously parse the output of my command. If it prints `a/b/c/d`, there are 3 ways to interpret it.